src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -88,10 +88,33 @@ class AppTasksResource @Inject() (service: MarathonSchedulerService,
                 @PathParam("taskId") id: String,
                 @QueryParam("scale") scale: Boolean = false): Response = {
     val pathId = appId.toRootPath
+
+    import mesosphere.util.ThreadPoolContext.context
+
     if (taskTracker.contains(pathId)) {
       val tasks = taskTracker.get(pathId)
       tasks.find(_.getId == id).fold(unknownTask(id)) { task =>
-        service.killTasks(pathId, Seq(task), scale)
```

Comment:
This is fairy similar to the `deleteMany` call above -- so you would need to adjust that two, I guess.

I do not like that we now have more non-API logic in here than before. Maybe introduce a new service? It might even be something specific as the `TaskKillService` or something like that this contains this logic and is also used by the `deleteMany` call.

We can look at other groupManager.update calls later...